modules/core/04-channel/keeper/events.go

```diff
@@ -380,12 +400,17 @@ func emitChannelUpgradeTimeoutEvent(ctx sdk.Context, portID string, channelID st
 			sdk.EventTypeMessage,
 			sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
 		),
-	})
+	}
 }
 
-// emitErrorReceiptEvent emits an error receipt event
-func emitErrorReceiptEvent(ctx sdk.Context, portID string, channelID string, currentChannel types.Channel, upgrade types.Upgrade, err error) {
-	ctx.EventManager().EmitEvents(sdk.Events{
+// emitChannelUpgradeTimeoutEvent emits an upgrade timeout event.
+func emitChannelUpgradeTimeoutEvent(ctx sdk.Context, portID string, channelID string, currentChannel types.Channel, upgrade types.Upgrade) {
```

Comment:
It seems like all these emitEvent functions take the same params and just call a different event creation "type". maybe we can just have these as one util function that takes a param from an enum listing all the different channel event types?